### PR TITLE
Fix SQLite database viewer

### PR DESCRIPTION
**Pull request by AprilC**
Fix bug  #554: sqlite database viewer is broken

SQLite database viewer is broken on iOS. All table in database viewer are empty. The issue is this statement for column names isn't returning any rows. We don't know the reason yet. But even if we can get the column names from the current api, we will run the query for 3 times. With this change, we added a new API: queryTable. This API only needs to query the table once and get everything.

### CODE REVIEW

File: Classes/GlobalStateExplorers/DatabaseBrowser/FLEXSQLiteDatabaseManager.m [hunk 1]
```diff
@@ -267,34 +272,34 @@ - (FLEXSQLResult *)errorResult:(NSString *)description {
...
 }
 
 - (id)objectForColumnIndex:(int)columnIdx stmt:(sqlite3_stmt*)stmt {
     int columnType = sqlite3_column_type(stmt, columnIdx);
-    
+
     switch (columnType) {
         case SQLITE_INTEGER:
-            return @(sqlite3_column_int64(stmt, columnIdx)).stringValue;
+            return [NSString stringWithFormat:@"%lld", sqlite3_column_int64(stmt, columnIdx)];
```
Comment (matrush) [inline Classes/GlobalStateExplorers/DatabaseBrowser/FLEXSQLiteDatabaseManager.m:284]: Is this change necessary?

File: Classes/GlobalStateExplorers/DatabaseBrowser/FLEXSQLiteDatabaseManager.m [hunk 2]
```diff
@@ -267,34 +272,34 @@ - (FLEXSQLResult *)errorResult:(NSString *)description {
...
         case SQLITE_FLOAT:
-            return  @(sqlite3_column_double(stmt, columnIdx)).stringValue;
+            return [NSString stringWithFormat:@"%f", sqlite3_column_double(stmt, columnIdx)];
```
Comment (matrush) [inline Classes/GlobalStateExplorers/DatabaseBrowser/FLEXSQLiteDatabaseManager.m:286]: Same for this one, is it necessary?

File: Classes/GlobalStateExplorers/DatabaseBrowser/FLEXSQLiteDatabaseManager.m (hunk 1 above)
Comment (AprilC) [reply Classes/GlobalStateExplorers/DatabaseBrowser/FLEXSQLiteDatabaseManager.m:284]: in some cases, i got an exception for this. it's safe to change to stringWithFormat

File: Classes/GlobalStateExplorers/DatabaseBrowser/FLEXSQLiteDatabaseManager.m (hunk 2 above)
Comment (AprilC) [reply Classes/GlobalStateExplorers/DatabaseBrowser/FLEXSQLiteDatabaseManager.m:286]: see above.

**Comment by matrush** [comment]
I took another look at this.
It seems this is an issue for certain sqlite versions (at least on 3.28 that I've tested).
See here for more reports:
https://sqlite.org/forum/info/2ca63507adffb34e
Here is another one (the last reply said it's patched in the latest version)
https://sqlite.org/forum/info/0e55af1e0b1823f7

So I agree given the `PRAGMA table_info(...)` has issues on certain versions, we should try to avoid using it if possible.

Back to the current commit, it won't work as the SQLite specific logic is brought to `FLEXTableListViewController` which is suppose to be generic as it supports other DBs as well. So I think we have to do some extra work here. The performance turning is good but it might not be straightforward to support that in `FLEXRealmDatabaseManager`.

File: Classes/GlobalStateExplorers/DatabaseBrowser/FLEXTableListViewController.m [hunk 3]
```diff
@@ -67,64 +67,71 @@ - (void)viewDidLoad {
...
+
     self.tables.selectionHandler = ^(FLEXTableListViewController *host, NSString *tableName) {
-        NSArray *rows = [host.dbm queryAllDataInTable:tableName];
-        NSArray *columns = [host.dbm queryAllColumnsOfTable:tableName];
-        NSArray *rowIDs = [host.dbm queryRowIDsInTable:tableName];
-        UIViewController *resultsScreen = [FLEXTableContentViewController
-            columns:columns rows:rows rowIDs:rowIDs tableName:tableName database:host.dbm
-        ];
-        [host.navigationController pushViewController:resultsScreen animated:YES];
+        FLEXSQLResult *result = [host.dbm queryTable:tableName];
```
Comment (matrush) [inline Classes/GlobalStateExplorers/DatabaseBrowser/FLEXTableListViewController.m:72]: `FLEXTableListViewController` supports more than one db. You cannot have `FLEXSQLResult` dependencies here.

**Review by matrush — CHANGES_REQUESTED** [review]
See my above comment for more details but it cannot be merged as is.

**Comment by NSExceptional** [comment]
I'll see if I can find time to look at this this week 🤔 thanks @matrush snd OP

File: Classes/GlobalStateExplorers/DatabaseBrowser/FLEXSQLiteDatabaseManager.m (hunk 1 above)
Comment (NSExceptional) [reply Classes/GlobalStateExplorers/DatabaseBrowser/FLEXSQLiteDatabaseManager.m:284]: Do you have a crash log for that? I'm highly skeptical that could ever result in an exception/crash; `@()` should accept any return value and properly wrap it in `NSNumber`

**Comment by NSExceptional** [comment]
Closing in favor of the other PR
